http/obmc_shell.hpp

```diff
@@ -0,0 +1,199 @@
+#pragma once
+
+#include <pty.h>
+#include <termios.h>
+
+#include <app.hpp>
+#include <boost/asio/posix/stream_descriptor.hpp>
+#include <boost/process/io.hpp>
+#include <websocket.hpp>
+
+#include <map>
+
+namespace crow
+{
+
+namespace obmc_shell
+{
+
+class Handler : public std::enable_shared_from_this<Handler>
+{
+  public:
+    Handler(crow::websocket::Connection* conn) :
+        session(conn), streamFileDescriptor(conn->getIoContext()),
+        doingWrite(false)
+    {}
+
+    ~Handler() = default;
+
+    void doClose()
+    {
+        streamFileDescriptor.close();
+
+        // boost::process::child::terminate uses SIGKILL, need to send SIGTERM
+        int rc = kill(pid, SIGKILL);
+        session = nullptr;
+        if (rc)
+        {
+            return;
+        }
+        waitpid(pid, nullptr, 0);
+    }
+
+    void connect()
+    {
+        pid = forkpty(&ttyFileDescriptor, nullptr, nullptr, nullptr);
+
+        if (pid == -1)
+        {
+            // ERROR
+            if (session)
+            {
+                session->close("Error creat child processSSH");
+            }
+            return;
+        }
+        if (pid == 0)
+        {
+            // CHILD
```

Comment:
At this point in the code flow (AFTER the child process is created and BEFORE the shell is exec'd) the process should change away from the "root" user to the "service" user.
If this is not done, the shell will run as the root user.  The correct behavior is for the shell to run as the user who invoked the URI, that is, the "service" user.
A testcase for this is to create a shell with this new function and invoke the command `echo $UID` to learn which user is running.  0 means root.  (Note you can see user ids in the /etc/passwd file.)  The "service" user has UID 1001.
I think this should be fixed before this is merged.

Reply:
code updated and tested